src/main/java/enmasse/kafka/bridge/tracker/FullOffsetTracker.java

```diff
@@ -40,15 +40,31 @@
 	// Apache Kafka topic to track
 	private String topic;
 	
-	// map with offset settlement status for each partition
-	private Map<Integer, Map<Long, Boolean>> offsetSettlements;
+	/**
+	 * The state of a partition
+	 */
+	private static class PartitionState {
+		/** Insertion-ordered map of offset to settlement */
+		public Map<Long, Boolean> settlements;
+		/** The current offset of this partition */
+		public long offset;
+		// has this message been received by remote AMQP peer
+		public boolean flag;
+		
+		public long firstUnsettled;
```

Comment:
In general I prefer to use private fields and getter/setter. What do you think about that ? Just to know your opinion ;)